### PR TITLE
Binary RValue

### PR DESCRIPTION
**Pull request by wemrysi**
Preliminary work for ch7173.

### CODE REVIEW

File: common/src/main/scala/quasar/common/data/RValue.scala [hunk 1]
```diff
@@ -378,8 +386,8 @@ sealed abstract class RValueInstances {
   implicit val show: Show[RValue] =
     Show.showFromToString
 
-  implicit val qdataEncode: QDataEncode[RValue] = QDataRValue
-  implicit val qdataDecode: QDataDecode[RValue] = QDataRValue
+  implicit def qdataEncode: QDataEncode[RValue] = QDataRValue
+  implicit def qdataDecode: QDataDecode[RValue] = QDataRValue
```
Comment (rintcius) [inline common/src/main/scala/quasar/common/data/RValue.scala]: Curious, why this change from `val` to `def`?

Comment (wemrysi) [reply common/src/main/scala/quasar/common/data/RValue.scala]: The impl of `QDataRValue` references some `vals` defined in the `RValue` companion leading to NPE due to initialization order.

File: common/src/main/scala/quasar/common/data/QDataRValue.scala [hunk 2]
```diff
@@ -173,16 +194,29 @@ object QDataRValue extends QDataEncode[RValue] with QDataDecode[RValue] {
 
   def getMetaValue(a: RValue): RValue = a match {
     case RMeta(value, _) => value
+    case CBinary(bytes) => RValue.rString(CBinary.toBase64String(bytes))
```
Comment (alissapajer) [inline common/src/main/scala/quasar/common/data/QDataRValue.scala]: Why the decision to turn the bytes into a string rather than keeping them as a byte array?
